From 779829b1eb9a673faf2709fca84bbcca75b9c4e5 Mon Sep 17 00:00:00 2001
From: leebob <leeboby@aliyun.com>
Date: Thu, 5 May 2022 11:24:47 +0800
Subject: [PATCH] fix h6 led

---
 board/sunxi/board.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/board/sunxi/board.c b/board/sunxi/board.c
index 28f702bc..cad84e34 100644
--- a/board/sunxi/board.c
+++ b/board/sunxi/board.c
@@ -627,6 +627,11 @@ void sunxi_board_init(void)
 {
  int power_failed = 0;
 
+#if defined(CONFIG_MACH_SUN50I_H6)
+ gpio_direction_output(SUNXI_GPL(4), 1);
+ gpio_direction_output(SUNXI_GPL(7), 0);
+#endif
+
 #ifdef CONFIG_LED_STATUS
  if (IS_ENABLED(CONFIG_SPL_DRIVERS_MISC))
   status_led_init();
-- 
2.35.1
