diff --git a/drivers/net/wireless/aic8800_sdio/aic8800_bsp/aicsdio.c b/drivers/net/wireless/aic8800_sdio/aic8800_bsp/aicsdio.c
index 3246be4..2bc5fbb 100644
--- a/drivers/net/wireless/aic8800_sdio/aic8800_bsp/aicsdio.c
+++ b/drivers/net/wireless/aic8800_sdio/aic8800_bsp/aicsdio.c
@@ -1377,7 +1377,7 @@ static void aicwf_sdio_bus_pwrctl(ulong data)
 	struct aic_sdio_dev *sdiodev = (struct aic_sdio_dev *) data;
 #endif
 	if (sdiodev->bus_if->state == BUS_DOWN_ST) {
-		sdio_err("bus down\n");
+		sdio_info("bus down\n");
 		return;
 	}
 
diff --git a/drivers/net/wireless/aic8800_sdio/aic8800_fdrv/rwnx_main.c b/drivers/net/wireless/aic8800_sdio/aic8800_fdrv/rwnx_main.c
index 8e3ba9c..989f195 100644
--- a/drivers/net/wireless/aic8800_sdio/aic8800_fdrv/rwnx_main.c
+++ b/drivers/net/wireless/aic8800_sdio/aic8800_fdrv/rwnx_main.c
@@ -3234,8 +3234,7 @@ static struct wireless_dev *rwnx_virtual_interface_add(struct rwnx_hw *rwnx_hw,
 
 	memcpy(vif->wdev.address, rwnx_hw->wiphy->perm_addr, ETH_ALEN);
 	vif->wdev.address[5] ^= vif_idx;
-	printk("p2p dev addr=%x %x %x %x %x %x\n", vif->wdev.address[0], vif->wdev.address[1], \
-		vif->wdev.address[2], vif->wdev.address[3], vif->wdev.address[4], vif->wdev.address[5]);
+	pr_debug("p2p dev addr=%pM\n", vif->wdev.address);
 
 	return wdev;
 }
